File: Algorithms/JavaScript/Search/recursiveBinarySearch.js
let arr = [2, 4, 5, 6, 7, 9, 10, 12, 15, 16]
let startIndex = 0
let endIndex = arr.length - 1

function recursiveBinarySearch(arr, startIndex, endIndex, target) {
  if (startIndex > endIndex) return -1 // Base case

  let midIndex = Math.floor((startIndex + endIndex) / 2)

  if (arr[midIndex] === target) return midIndex

  if (arr[midIndex] > target) return recursiveBinarySearch(arr, startIndex, midIndex - 1, target)
  else return recursiveBinarySearch(arr, midIndex + 1, endIndex, target)
}

// Test Cases:

let target1 = 15
console.log(recursiveBinarySearch(arr, startIndex, endIndex, target1)) // returns 8, the index of target

let target2 = 8
console.log(recursiveBinarySearch(arr, startIndex, endIndex, target2)) // returns -1, target not found

File: Algorithms/Search/Python/recursiveBinarySearch.py
import math

arr = [1, 4, 6, 7, 12, 32, 47, 54, 324]
startIndex = 0
endIndex = len(arr) - 1

def binarySearch(arr, startIndex, endIndex, target):
  if(startIndex > endIndex): return -1

  midIndex = math.floor((startIndex + endIndex) / 2)

  if(target == arr[midIndex]): return midIndex
  if(target < arr[midIndex]): return binarySearch(arr, startIndex, midIndex - 1, target)
  else: return binarySearch(arr, midIndex + 1, endIndex, target)

# Test Cases:

target1 = int(3)
print(binarySearch(arr, startIndex, endIndex, target1)) # returns -1, target not found

target2 = int(47)
print(binarySearch(arr, startIndex, endIndex, target2)) # returns 6, the index of target